src/Symfony/Component/Lock/Store/StoreFactory.php

```diff
@@ -48,6 +48,9 @@ public static function createStore($connection)
             case $connection instanceof \Memcached:
                 return new MemcachedStore($connection);
 
+            case $connection instanceof \MongoDB\Collection:
```

Comment:
Missing cases:
- `$connection instanceof \MongoDB\Client`
- `0 === strpos($connection, 'mopngodb://')`

Reply:
This is not possible without the mandatory `$options['collection']` and `$options['database']`.

It's not mandatory or even possible in a MongoDB Connection URI to specify either of these. Since `StoreFactory` does not support `$options` it's therefore only possible to use it to construct a `MongoDbStore` using a `MongoDB\Collection` where the options are note required / used.

Reply:
Due to new support for `/database?collection=` in Connection String URIs i've added support for `0 === strpos($connection, 'mongodb')`.

Please note, I have intentionally omitted `://` to allow DNS Seed list connection strings: https://docs.mongodb.com/manual/reference/connection-string/#dns-seedlist-connection-format

We still cannot support `$connection instanceof \MongoDB\Client` due to the requirements for `$options['collection']` and `$options['database']`.